src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1345 @@
+            try {
+                $frameQuery = $this->webDriver->findElement(WebDriverBy::id($name));
+            } catch (NoSuchElementException $e) {
+                $frameQuery = $this->webDriver->findElement(WebDriverBy::name($name));
+            }
+        }
+
+        $this->webDriver->switchTo()->frame($frameQuery);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCookie($name, $value = null): void
+    {
+        if (null === $value) {
+            $this->webDriver->manage()->deleteCookieNamed($name);
+
+            return;
+        }
+
+        $cookieArray = [
+            'name' => $name,
+            'value' => rawurlencode($value),
+            'secure' => false,
+        ];
+
+        $this->webDriver->manage()->addCookie($cookieArray);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCookie($name): ?string
+    {
+        try {
+            $result = $this->webDriver->manage()->getCookieNamed($name);
+        } catch (NoSuchCookieException $e) {
+            $result = null;
+        }
+        if ($result === null) {
+            return null;
+        }
+
+        $result = $result->getValue();
+        if ($result === null) {
+            return null;
+        }
+
+        return rawurldecode($result);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContent(): string
+    {
+        return $this->webDriver->getPageSource();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getScreenshot(): string
+    {
+        return $this->webDriver->takeScreenshot();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getWindowNames(): array
+    {
+        $origWindow = $this->webDriver->getWindowHandle();
+
+        try {
+            $result = [];
+            foreach ($this->webDriver->getWindowHandles() as $tempWindow) {
+                $this->webDriver->switchTo()->window($tempWindow);
+                $result[] = $this->getWindowName();
+            }
+            return $result;
+        } finally {
+            $this->webDriver->switchTo()->window($origWindow);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getWindowName(): string
+    {
+        $name = (string)$this->evaluateScript('window.name');
+
+        if ($name === '') {
+            $name = self::W3C_WINDOW_HANDLE_PREFIX . $this->webDriver->getWindowHandle();
+        }
+
+        return $name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findElementXpaths(
+        #[Language('XPath')]
+        $xpath
+    ): array {
+        $nodes = $this->webDriver->findElements(WebDriverBy::xpath($xpath));
+
+        $elements = [];
+        foreach ($nodes as $i => $node) {
+            $elements[] = sprintf('(%s)[%d]', $xpath, $i + 1);
+        }
+
+        return $elements;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws NoSuchElementException
```

Comment:
this is wrong. the interface says that methods should throw a DriverException when the operation cannot be done.